src/aggregation/mod.rs

```diff
@@ -239,6 +245,14 @@ pub enum Key {
     /// `f64` key
     F64(f64),
 }
+impl From<IntermediateKey> for Key {
```

Comment:
nitpick It woudl have been more inspired to put that in 
intermediate_agg_result.rs,
with the other conversion.